src/Raven.Server/Indexing/LuceneVoronDirectory.cs

```diff
@@ -68,19 +68,18 @@ public override void TouchFile(string name)
             if (readResult == null)
                 throw new FileNotFoundException("Could not find file", name);
 
-            filesTree.Add(name, Stream.Null);
+            var pos = filesTree.DirectAdd(name, readResult.Reader.Length);
```

Comment:
Version will be modified underneath. And we use that value in `FileModified` method.